tensorflow/python/autograph/operators/py_builtins.py

```diff
@@ -384,8 +384,29 @@ def _py_filter(function, iterable):
   return filter(function, iterable)
 
 
+def any_(iterable):
+  if isinstance(iterable, dataset_ops.DatasetV2):
+    return _tf_dataset_any(iterable)
+  return _py_any(iterable)
+
+
+# any() operation is essentially a "if first True element exist".
+# For that it could be translated to `filter(True)` to filter out
+# only `True` element, and then `take(1)`. This works in tf.data
+# as tf.data's filter+take is done in pipeline so it will stop
+# as soon as `take(1)` returns.
+def _tf_dataset_any(iterable):
```

Comment:
A great addition would be a verification that looks at `iterable.element_spec` and raises an error if the dataset contains anything other than a single `bool` scalar tensor. The error message can be something like this:  `in graph mode, the "any" builtin only supports datasets that return bool scalars; got: {}`.format(iterable.element_spec)

Reply:
Thanks. The PR has been updated to check and makes sure element_spec only consists of tf.bool.